feat(linter/no-extraneous-class): add conditional fixer

DonIsaac opened the pull request:
## What This PR Does
Adds a `suggestion`-level auto-fixer to remove empty classes. Empty classes with decorators are not removed.

DonIsaac commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10798?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10798** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10798?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10798?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10797** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10797?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10798).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_linter/src/rules/typescript/no_extraneous_class.rs:
```diff
@@ -320,5 +328,17 @@ fn test() {
 
-    Tester::new(NoExtraneousClass::NAME, NoExtraneousClass::PLUGIN, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("class Foo {}", "", None, FixKind::Suggestion),
```
Can you please add a test for `export class Foo {}` too?
I guess this will result into `export` and will lead into invalid syntax.
Deleting the export will not be safe because other files could use the import.

Maybe we should mark them as dangerous suggestions when they have an export parent?

camc314 replied:
I have made it always a dangerous suggestion, and fixed the `export class` case

If a user had code that referenced this class, removeing the class (even if there's no export) is dangrous, as the IdentiferRefernece now refernces a symbol that no longer exists.

DonIsaac committed:
feat(linter/no-extraneous-class): add conditional fixer (#10798)

## What This PR Does
Adds a `suggestion`-level auto-fixer to remove empty classes. Empty classes with decorators are not removed.
